### PR TITLE
docs: added docs on localization & culture hotswapping

### LINKED ISSUE #5437: Docs: How-to and sample for localization and runtime culture switching
## Concern?

- [ ] Usage in industry
- [ ] Clarification of capabilities
- [ ] Getting started with Uno
- [ ] Developing with Uno
- [ ] Contributing to the Uno project
- [ ] Publishing your application
- [ ] Support
- [ ] Other (please specify):

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable):
resolve #5437

## PR Type
What kind of change does this PR introduce?
- Documentation content changes

## What is the new behavior?
added docs on localization & culture hotswapping

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
Related sample code is in another PR: unoplatform/Uno.Samples#47

### CODE REVIEW

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# How to use localization and perform runtime culture switching
+## Full Sample
+Both complete sample code can be found here:
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:3]: Modify the intro to match the template: https://github.com/unoplatform/uno/blame/dev/djo/docs-fix-3/doc/.howto-template.md

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+- [Localization](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization)
+- [RuntimeCultureSwitching](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching)
+
+## Localization
+
+1. Create a new Uno Platform application, following the instructions [here](..\get-started.md).
+1. Modify the content of `MainPage` a bit:
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:10]:
```suggestion
1. Modify the content of `MainPage`:
```

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+              xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+              xmlns:toolkit="using:Uno.UI.Toolkit">
+
+            <StackPanel toolkit:VisibleBoundsPadding.PaddingMask="Top">
+                <TextBlock x:Uid="MainPage_IntroText" Text="Hello, world!" Margin="20" FontSize="30" />
+                <TextBlock x:Name="CodeBehindText" Text="This text will be replaced" />
+            </StackPanel>
+        </Page>
+        ```
+        > note:
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:25]:
```suggestion
        > [!NOTE]
```

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        ```cs
+        public MainPage()
+        {
+            this.InitializeComponent();
+            this.Loaded += (s, e) => CodeBehindText.Text = Windows.ApplicationModel.Resources.ResourceLoader
+                .GetForViewIndependentUse()
+                .GetString("MainPage_CodeBehindString");
+        }
+        ```
+1. Create a new resource file for localization in french in the `UnoLocalization.Shared` project:
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:40]:
```suggestion
1. Create a new resource file for localization in French in the `UnoLocalization.Shared` project:
```
```suggestion
1. Create a new resource file for localization in french in the `UnoLocalization.Shared` project:
```

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    1. Add a new folder `fr` under the `Strings` folder by:
+    R-Click on `String` > Add > New Folder
+    1. Add a new resource file `Resources.resw` under the `fr` folder by:
+    R-Click on `fr` > Add > New Item ... > Visual C# > Xaml > Resources File
+1. Add some localization texts for both english and french:
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:45]:
```suggestion
1. Add some localized text entries for both English and French:
```

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+            this.InitializeComponent();
+            this.Suspending += OnSuspending;
+        }
+
+        private void ChangeStartingLanguage()
+        {
+            var culture = new System.Globalization.CultureInfo("fr");
+
+        #if WINDOWS_UWP
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:74]: per the discussion on the other PR, I think you can safely use this Uno 3.6 feature

File: doc/articles/guides/localization-and-culture-hot-swapping.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride = culture.IetfLanguageTag;
+        #else
+            System.Globalization.CultureInfo.CurrentUICulture = culture;
+            System.Globalization.CultureInfo.CurrentCulture = culture;
+            Windows.ApplicationModel.Resources.ResourceLoader.DefaultLanguage = culture.IetfLanguageTag;
+        #endif
+        }
+        ```
+
+## Runtime culture switching
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization-and-culture-hot-swapping.md:84]: Put this in a separate article

**Review by carldebilly — APPROVED** [review]

File: doc/articles/guides/localization-and-culture-hot-swapping.md (hunk 7 above)
Comment (Xiaoy312) [reply doc/articles/guides/localization-and-culture-hot-swapping.md]: done

File: doc/articles/guides/hotswap-app-language.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
+# How to change app language at runtime
+
+This guide will walk you through the necessary steps for changing app language at runtime.
+
+> [!TIP] The complete source code that goes along with this guide is available in the [unoplatform/Uno.Samples](https://github.com/unoplatform/Uno.Samples) GitHub repository - [RuntimeCultureSwitching](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching)
+
+## Prerequisites
```
Comment (davidjohnoliver) [inline doc/articles/guides/hotswap-app-language.md:7]: Copy-paste the 'Prerequisites' section verbatim from the template.

File: doc/articles/guides/hotswap-app-language.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
+# How to change app language at runtime
+
+This guide will walk you through the necessary steps for changing app language at runtime.
+
+> [!TIP] The complete source code that goes along with this guide is available in the [unoplatform/Uno.Samples](https://github.com/unoplatform/Uno.Samples) GitHub repository - [RuntimeCultureSwitching](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching)
```
Comment (davidjohnoliver) [inline doc/articles/guides/hotswap-app-language.md:5]:
```suggestion
> [!TIP]
> The complete source code that goes along with this guide is available in the [unoplatform/Uno.Samples](https://github.com/unoplatform/Uno.Samples) GitHub repository - [RuntimeCultureSwitching](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching)
```

File: doc/articles/guides/hotswap-app-language.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+* [**Rider Version 2020.2+**](https://www.jetbrains.com/rider/download/)
+* [**Rider Xamarin Android Support Plugin**](https://plugins.jetbrains.com/plugin/12056-rider-xamarin-android-support/) (you may install it directly from Rider)
+
+## Step-by-steps
+> [!NOTE]  
+> This guide is an extension of ["How to use localization"](localization.md), and will build on top the sample from that guide.
+> Make sure you have completed the previous guide or downloaded the [full localization sample](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization), before continuing.
+
+1. Add two new pages to the `Localization.Shared` project by:
+    R-Click on `Localization.Shared` > Add > New Item ... > Visual C# > XAML > Blank Page
```
Comment (davidjohnoliver) [inline doc/articles/guides/hotswap-app-language.md:50]:
```suggestion
    Right-click on `Localization.Shared` > Add > New Item ... > Visual C# > XAML > Blank Page
```

File: doc/articles/guides/hotswap-app-language.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+    -|-|-
+    MainPage_GotoNextPage.Content|Next Page|Page suivante
+    MainPage_GotoLanguageSettings.Content|Language Settings|Paramètres de langue
+    Page1_Title.Text|Page in english|Page en français
+    Page1_GoBack.Content|Go back|Retourner
+    LanguageSettings_Title.Text|Language Settings|Paramètres de langue
+    LanguageSettings_GoBack.Content|Go back|Retourner
+
+## Additional Resources
+https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/globalizing-portal
```
Comment (davidjohnoliver) [inline doc/articles/guides/hotswap-app-language.md:139]:
```suggestion
## Get the complete code

See the completed sample on GitHub: [RuntimeCultureSwitching](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching)

## Additional Resources
https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/globalizing-portal

<br>

***

## Help! I'm having trouble

> [!TIP]
> If you ran into difficulties with any part of this guide, you can:
> 
> * Ask for help on our [Discord channel](https://www.platform.uno/discord) - #uno-platform
> * Ask a question on [Stack Overflow](https://stackoverflow.com/questions/tagged/uno-platform) with the 'uno-platform' tag
```

File: doc/articles/guides/localization.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
+# How to use localization
+
+This guide will walk you through the necessary steps to localize an Uno Platform application.
+
+> [!TIP] The complete source code that goes along with this guide is available in the [unoplatform/Uno.Samples](https://github.com/unoplatform/Uno.Samples) GitHub repository - [Localization](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization)
+
+## Prerequisites
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization.md:7]: Copy from the template.

File: doc/articles/guides/localization.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            this.InitializeComponent();
+            this.Loaded += (s, e) => CodeBehindText.Text = Windows.ApplicationModel.Resources.ResourceLoader
+                .GetForViewIndependentUse()
+                .GetString("MainPage_CodeBehindString");
+        }
+        ```
+1. Create a new resource file for localization in French in the `UnoLocalization.Shared` project:
+    1. Add a new folder `fr` under the `Strings` folder by:
+    R-Click on `String` > Add > New Folder
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization.md:78]:
```suggestion
    Right-click on `String` > Add > New Folder
```

File: doc/articles/guides/localization.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+    1. Add a new resource file `Resources.resw` under the `fr` folder by:
+    R-Click on `fr` > Add > New Item ... > Visual C# > Xaml > Resources File
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization.md:80]:
```suggestion
    Right-click on `fr` > Add > New Item ... > Visual C# > Xaml > Resources File
```

File: doc/articles/guides/localization.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
+# How to use localization
+
+This guide will walk you through the necessary steps to localize an Uno Platform application.
+
+> [!TIP] The complete source code that goes along with this guide is available in the [unoplatform/Uno.Samples](https://github.com/unoplatform/Uno.Samples) GitHub repository - [Localization](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization)
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization.md:5]:
```suggestion
> [!TIP] 
> The complete source code that goes along with this guide is available in the [unoplatform/Uno.Samples](https://github.com/unoplatform/Uno.Samples) GitHub repository - [Localization](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization)
```

File: doc/articles/guides/localization.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+        private void ChangeStartingLanguage()
+        {
+            var culture = new System.Globalization.CultureInfo("fr");
+
+            Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride = culture.IetfLanguageTag;
+        }
+        ```
+
+## Additional Resources
+https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/globalizing-portal
```
Comment (davidjohnoliver) [inline doc/articles/guides/localization.md:115]:
```suggestion
## Get the complete code

See the completed sample on GitHub: [LocalizationSamples/Localization](https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/Localization)

## Additional Resources
https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/globalizing-portal

<br>

***

## Help! I'm having trouble

> [!TIP]
> If you ran into difficulties with any part of this guide, you can:
> 
> * Ask for help on our [Discord channel](https://www.platform.uno/discord) - #uno-platform
> * Ask a question on [Stack Overflow](https://stackoverflow.com/questions/tagged/uno-platform) with the 'uno-platform' tag
```

File: doc/articles/toc.yml [hunk 17]
```diff
@@ -38,6 +38,10 @@
           href: splash-screen.md
         - name: How to consume a web API
           href: howto-consume-webservices.md
+        - name: How to use localization
```
Comment (davidjohnoliver) [inline doc/articles/toc.yml:41]:
```suggestion
        - name: How to localize text resources
```

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**